core/src/services/web-components.js

```diff
@@ -309,22 +430,57 @@ class WebComponentSvcClass {
       };
     } else if (navNode.compound.renderer) {
       renderer = resolveRenderer(navNode.compound.renderer);
+    } else {
+      renderer = new DefaultCompoundRenderer();
     }
 
-    renderer = renderer || new DefaultCompoundRenderer();
+    return renderer;
+  }
+
+  createIntersectionObserver(navNode) {
+    return new IntersectionObserver(
+      (entries, observer) => {
+        this.intersectionObserverCallback(entries, observer);
+      },
+      {
+        rootMargin:
+          navNode.compound.lazyLoadingOptions?.intersectionRootMargin || DEFAULT_INTERSECTION_OBSERVER_ROOTMARGIN
+      }
+    );
+  }
+
+  /**
+   * Responsible for rendering web component compounds based on a renderer or a nesting
+   * micro frontend.
+   *
+   * @param {*} navNode the navigation node defining the compound
+   * @param {*} wc_container the web component container dom element
+   * @param {*} context the luigi node context
+   */
+  renderWebComponentCompound(navNode, wc_container, extendedContext) {
+    const useLazyLoading = navNode.compound?.lazyLoadingOptions?.enabled === true;

```

Comment:
Can you please document this settings object `lazyLoadingOptions`.
E.g [here](https://github.com/SAP/luigi/blob/main/docs/navigation-parameters-reference.md#compound) and maybe it is worth to mention this object in this [document](https://github.com/SAP/luigi/blob/main/docs/web-component.md#compound-web-components) as well.

Reply:
Done...